### PR TITLE
Alerting: app specific query editors for Loki and Prometheus

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
This PR will render a simplified version of the query editor if we are in the alerting context. When creating a cloud alerting rule we only want to specify the `expr` field and no other options should be available.

**Special notes for your reviewer**:
I had a chat with @torkelo on how to approach this and we decided that introducing an app flag and render different editors based on that is the way to go.

### CODE REVIEW

File: packages/grafana-data/src/types/app.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export enum CoreApp {
   Dashboard = 'dashboard',
   Explore = 'explore',
   Unknown = 'unknown',
+  CloudAlerting = 'cloud-alerting',
```
Comment (torkelo) [inline packages/grafana-data/src/types/app.ts:10]: should there not be an Alerting option here as well?

**Review by torkelo — APPROVED** [review]

**Review by peterholmberg — COMMENTED** [review]
Super nice!

Could we update this placeholder text since shift enter does not actually run the query. 
Screenshot 2021-05-19 112838

File: packages/grafana-data/src/types/app.ts (hunk 1 above)
Comment (mckn) [reply packages/grafana-data/src/types/app.ts:10]: Not yet, but we will probably add it in PRs to come.

**Comment by mckn** [comment]
> Super nice!
> 
> Could we update this placeholder text since shift enter does not actually run the query.
> Screenshot 2021-05-19 112838

Yupp, fixing that one. Great feedback!

File: public/app/plugins/datasource/prometheus/components/PromQueryField.tsx [hunk 2]
```diff
@@ -77,6 +77,8 @@ export function willApplySuggestion(suggestion: string, { typeaheadContext, type
 interface PromQueryFieldProps extends ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions> {
   history: Array<HistoryItem<PromQuery>>;
   ExtraFieldElement?: ReactNode;
+  placeholder?: string;
+  'data-testid'?: string;
```
Comment (mckn) [inline public/app/plugins/datasource/prometheus/components/PromQueryField.tsx:81]: Do you know if there is there a better way of doing this? @domasx2 @torkelo @hugohaggmark

Comment (torkelo) [reply public/app/plugins/datasource/prometheus/components/PromQueryField.tsx:81]: No, don't think there is

Comment (mckn) [reply public/app/plugins/datasource/prometheus/components/PromQueryField.tsx:81]: Ok, thanks 🙏🏻

Comment (hugohaggmark) [reply public/app/plugins/datasource/prometheus/components/PromQueryField.tsx:81]: Shouldn't we always use `aria-label`? Then we get accessibility and testing in the same prop?

**Pull request merged**